### PR TITLE
Tensorpack Update and OpenCv dependency

### PR DESCRIPTION
**Pull request by rizasif**
Hi,

The repository in the current form was not functional due to an update with [Tensorpack](https://github.com/ppwwyyxx/tensorpack). This pull request is inspired by the issue https://github.com/ppwwyyxx/tensorpack/issues/458. Moreover, it was found that OpenCv-Python is actually required for execution, hence, it was added as dependency in the ReadMe file.

I have made the necessary changes and tried them on my server with Ubuntu 17.04, Nvidia GTX-1050 and Cuda 8.0. It appears to be working fine now.

Please update.

Thank you.

### CODE REVIEW

**Comment by rizasif** [comment]
Here is the error received if Tensorpack update not included.

```
[0317 16:22:46 @logger.py:119] Use a new log directory train_log/cifar10-single-fisrt150-second225-max3000317-162246
[0317 16:22:46 @logger.py:74] Argv: cifar10-densenet.py --gpu=1
[0317 16:22:46 @fs.py:89] WRN Env var $TENSORPACK_DATASET not set, using /home/turanshare/tensorpack_data for datasets.
[0317 16:22:46 @cifar.py:33] Found cifar10 data in /home/turanshare/tensorpack_data/cifar10_data.
[0317 16:22:47 @parallel.py:178] [MultiProcessPrefetchData] Will fork a dataflow more than one times. This assumes the datapoints are i.i.d.
[0317 16:22:47 @cifar.py:33] Found cifar10 data in /home/turanshare/tensorpack_data/cifar10_data.
[0317 16:22:47 @base.py:344] WRN You're calling new trainers with old trainer API!
[0317 16:22:47 @base.py:345] WRN Now it returns the old trainer for you, please switch to use new trainers soon!
[0317 16:22:47 @base.py:346] WRN See https://github.com/ppwwyyxx/tensorpack/issues/458 for more information.
Traceback (most recent call last):
  File "cifar10-densenet.py", line 179, in <module>
    launch_train_with_config(config, SyncMultiGPUTrainer(config))
  File "/home/turanshare/anaconda2/envs/tensorflow/lib/python3.6/site-packages/tensorpack/train/trainers.py", line 104, in SyncMultiGPUTrainer
    return SyncMultiGPUTrainerParameterServer(gpus, ps_device='cpu')
  File "/home/turanshare/anaconda2/envs/tensorflow/lib/python3.6/site-packages/tensorpack/train/base.py", line 347, in __new__
    return old_trainer(*args, **kwargs)
  File "/home/turanshare/anaconda2/envs/tensorflow/lib/python3.6/site-packages/tensorpack/trainv1/multigpu.py", line 61, in __init__
    apply_prefetch_policy(config, gpu_prefetch)
  File "/home/turanshare/anaconda2/envs/tensorflow/lib/python3.6/site-packages/tensorpack/trainv1/multigpu.py", line 42, in apply_prefetch_policy
    if len(config.tower) > 1 and gpu_prefetch:
AttributeError: 'TrainConfig' object has no attribute 'tower'
```

**Comment by rizasif** [comment]
The OpenCv output can be viewed here https://github.com/YixuanLi/densenet-tensorflow/issues/15#issuecomment-373933248

File: cifar10-densenet.py [hunk 1]
```diff
@@ -173,6 +173,11 @@ def get_config():
     config = get_config()
     if args.load:
         config.session_init = SaverRestore(args.load)
+    
     if args.gpu:
         config.nr_tower = len(args.gpu.split(','))
-    SyncMultiGPUTrainer(config).train()
+    else:
+        config.nr_tower = 0
```
Comment (ppwwyyxx) [inline cifar10-densenet.py]: Setting `config.nr_tower` has no effect any more.
Using `SyncMultiGPUTrainer(num_gpu)` is sufficient.

**Comment by rizasif** [comment]
@ppwwyyxx please check now

**Comment by ppwwyyxx** [comment]
Looks good!

**Comment by rizasif** [comment]
Thank you :)

**Pull request merged**